core/main.go

```diff
@@ -562,36 +555,51 @@ func (t *Textile) FlushBlocks() {
 		return util.ProtoTime(pending.Items[i].Date).Before(
 			util.ProtoTime(pending.Items[j].Date))
 	})
-	var posted bool
+	wg := sync.WaitGroup{}
 	for _, block := range pending.Items {
 		if t.datastore.CafeRequests().SyncGroupComplete(block.Id) {
-			thread := t.Thread(block.Thread)
-			if thread == nil {
-				continue
-			}
+			wg.Add(1)
+			go func(block *pb.Block) {
+				var posted bool
+				thread := t.Thread(block.Thread)
+				if thread == nil {
+					return
+				}
 
-			err := thread.post(block)
-			if err != nil {
-				log.Errorf("error posting block %s: %s", block.Id, err)
-				continue
-			}
-			posted = true
+				err := thread.post(block)
+				if err != nil {
+					log.Errorf("error posting block %s: %s", block.Id, err)
+					if block.Attempts+1 >= maxDownloadAttempts {
```

Comment:
Adds the attempt handling used when a block is "pending" to when a block is "queued" - this way we can eventually stop trying to post the block.